packages/core/src/connectors/coinbaseWallet.ts

```diff
@@ -135,8 +129,20 @@ export class CoinbaseWalletConnector extends Connector<
         CoinbaseWalletSDK = (<{ default: typeof CoinbaseWalletSDK }>(
           (<unknown>CoinbaseWalletSDK)
         )).default
-
       this.#client = new CoinbaseWalletSDK(this.options)
+
+      // @ts-expect-error – accessing `private` method
+      const walletExtensionChainId = this.#client.walletExtension?.getChainId()
```

Comment:
Could explicitly type this instead of using `@ts-expect-error`. Not sure which one is better though.

```suggestion
      // Mock implementations for typing
      class WalletProvider {
        // https://github.com/coinbase/coinbase-wallet-sdk/blob/master/packages/wallet-sdk/src/provider/CoinbaseWalletProvider.ts#L927-L936
        getChainId(): number { return 1 }
      }
      class Client {
        // https://github.com/coinbase/coinbase-wallet-sdk/blob/master/packages/wallet-sdk/src/CoinbaseWalletSDK.ts#L233-L235
        get walletExtension(): WalletProvider | undefined { return }
      }
      const walletExtensionChainId = (this.#client as unknown as Client).walletExtension?.getChainId()
```

Reply:
Probably better than `@ts-expect-error`!